client/lib/feed-url-store/actions.js

```diff
@@ -0,0 +1,24 @@
+var inflight = require( 'lib/inflight' ),
+	wpcom = require( 'lib/wp' );
+
+function requestKey( feedId ) {
+	return `feed-${feedId}`;
+}
+
+const FeedUrlStoreActions = {
+	discover: function( feedUrl, fn ) {
+		const key = requestKey( feedUrl );
+
+		if ( inflight.requestInflight( key ) ) {
+			return;
```

Comment:
Good call, thanks. Working with promises makes the module really easy to use, and allows us to expose only one method, encapsulating cache and internal actions within it. This has been done in the new `feed-lookup` module.